core/client/src/main/java/alluxio/client/netty/NettyUnderFileSystemBlockReader.java

```diff
@@ -80,48 +70,45 @@ public ByteBuffer read(InetSocketAddress address, long ufsFileId, long offset, l
 
-      ChannelFuture channelFuture =
-          channel.writeAndFlush(new RPCFileReadRequest(ufsFileId, offset, length)).sync();
-
+      ChannelFuture channelFuture = channel.writeAndFlush(
+          new RPCUnderFileSystemBlockReadRequest(blockId, offset, length, sessionId, noCache));
+      channelFuture = channelFuture.sync();
       if (channelFuture.isDone() && !channelFuture.isSuccess()) {
-        LOG.error("Failed to read ufs file from %s for ufsFilId %d with error %s.",
-            address.toString(), ufsFileId, channelFuture.cause());
+        LOG.error("Failed to read from %s for block %d with error %s.", address.toString(), blockId,
+            channelFuture.cause());
         throw new IOException(channelFuture.cause());
       }
 
       RPCResponse response = listener.get(NettyClient.TIMEOUT_MS, TimeUnit.MILLISECONDS);
 
       switch (response.getType()) {
-        case RPC_FILE_READ_RESPONSE:
-          RPCFileReadResponse resp = (RPCFileReadResponse) response;
-          LOG.debug("Data for ufs file id {} from machine {} received", ufsFileId, address);
-          RPCResponse.Status status = resp.getStatus();
+        case RPC_BLOCK_READ_RESPONSE:
+          RPCBlockReadResponse blockResponse = (RPCBlockReadResponse) response;
+          LOG.debug("Data {} from machine {} received", blockId, address);
+
+          RPCResponse.Status status = blockResponse.getStatus();
           if (status == RPCResponse.Status.SUCCESS) {
             // always clear the previous response before reading another one
-            cleanup();
-            // End of file reached
-            if (resp.isEOF()) {
-              return null;
-            }
-            mReadResponse = resp;
-            return resp.getPayloadDataBuffer().getReadOnlyByteBuffer();
+            close();
+            mReadResponse = blockResponse;
+            return blockResponse.getPayloadDataBuffer().getReadOnlyByteBuffer();
           }
-          throw new IOException(status.getMessage() + " response: " + resp);
+          throw new IOException(status.getMessage() + " response: " + blockResponse);
         case RPC_ERROR_RESPONSE:
           RPCErrorResponse error = (RPCErrorResponse) response;
           throw new IOException(error.getStatus().getMessage());
         default:
-          throw new IOException(ExceptionMessage.UNEXPECTED_RPC_RESPONSE.getMessage(
-              response.getType(), RPCMessage.Type.RPC_FILE_READ_RESPONSE));
+          throw new IOException(ExceptionMessage.UNEXPECTED_RPC_RESPONSE
+              .getMessage(response.getType(), RPCMessage.Type.RPC_BLOCK_READ_RESPONSE));
       }
     } catch (Exception e) {
-      Metrics.NETTY_UFS_READ_FAILURES.inc();
+      Metrics.NETTY_UFS_BLOCK_READ_FAILURES.inc();
       try {
         if (channel != null) {
           channel.close().sync();
         }
       } catch (InterruptedException ee) {
-        Throwables.propagate(ee);
+        throw Throwables.propagate(ee);
```

Comment:
FYI, `Throwables.propagate` is deprecated and `new RuntimeException` should be used instead (https://github.com/google/guava/issues/1366)